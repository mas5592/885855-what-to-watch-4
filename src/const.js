export const ENTRY_POINT = `https://4.react.pages.academy/wtw`;

export const AVATAR_URL = `https://4.react.pages.academy/wtw/asda.jpg`;

export const TIMEOUT_DURATION = 5000;

export const MAX_SIMILAR_CARDS = 4;

export const LIMIT_FILMS_COUNT = 8;

export const RATINGS_NUMBER = 5;

export const FILTER_ALL_GENRES = `All genres`;

export const PageNames = {
  MAIN: `main`,
  FILM_DETAILS: `Details`,
  SIGN_IN: `sign in`,
  ADD_REVIEW: `add review`,
  MY_LIST: `my list`,
};

export const AppRoute = {
  LOGIN: `/login`,
  ROOT: `/`,
  PAGE: `/film`,
  FILM_REVIEW: `/review`,
  MY_LIST: `/mylist`,
  VIDEO_PLAYER: `/player`,
};

export const RatingLevel = {
  AWESOME: `Awesome`,
  VERY_GOOD: `Very good`,
  GOOD: `Good`,
  NORMAL: `Normal`,
  BAD: `Bad`
};

export const ListFilters = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`
};

export const Time = {
  MINUTES_IN_HOUR: 60,
  SECONDS_IN_MINUTE: 60,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const ReviewLength = {
  MIN: 50,
  MAX: 400,
};

export const reviewSubmitButton = {
  post: `Post`,
  sending: `Sending...`,
};

export const emptyFilm = {
  title: `Loading...`,
  genre: ``,
  date: 0,
  coverBackground: ``,
  poster: ``,
  id: 0,
  description: ``,
  rating: 0,
  votes: 0,
  director: ``,
  starring: [],
  runTime: 0,
  preview: ``,
  videoLink: ``,
  isFavorite: false,
  backgroundColor: ``,
};

export const activeGenre = `Drama`;

export const FormatDate = {
  DATA: `MMMM D, YYYY`,
  DATA_TO_ISO: `YYYY-MM-DD`,
};
