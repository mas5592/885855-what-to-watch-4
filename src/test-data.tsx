import {FilmType, ReviewType} from './types';

export const card: FilmType = {
  title: `Name`,
  genre: `film Genre`,
  date: 1993,
  coverBackground: `https://placeimg.com/1300/512/nature`,
  poster: `img/snatch.jpg`,
  picture: `img/snatch.jpg`,
  id: 123456,
  description: `Place to describe the movie`,
  rating: 8.9,
  scores: 240,
  director: `Director Name`,
  starring: [`Author`],
  preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  filmDurationTime: 107,
  videoUrl: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  isFavorite: false,
  backgroundColor: `#FFFFFF`,
};

export const films: Array<FilmType> = [
  {
    title: `Name`,
    genre: `Genre`,
    date: 2014,
    coverBackground: `https://placeimg.com/1300/512/nature`,
    poster: `img/snatch.jpg`,
    picture: `img/snatch.jpg`,
    id: 123465,
    description: `Place to describe the movie`,
    rating: 8.3,
    scores: 1500,
    director: `Director Name`,
    starring: [`Author`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    filmDurationTime: 107,
    videoUrl: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    isFavorite: false,
    backgroundColor: `#FFFFFF`,
  },
  {
    title: `Name`,
    genre: `Genre`,
    date: 2014,
    coverBackground: `https://placeimg.com/1300/512/nature`,
    poster: `img/snatch.jpg`,
    picture: `img/snatch.jpg`,
    id: 123478,
    description: `film description.`,
    rating: 8.3,
    scores: 1500,
    director: `Director Name`,
    starring: [`Author`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    videoUrl: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    filmDurationTime: 107,
    isFavorite: false,
    backgroundColor: `#FFFFFF`,
  }
];

export const activeFilm: FilmType = {
  title: `Snatch`,
  genre: `Genre`,
  date: 2014,
  coverBackground: `https://placeimg.com/1300/512/nature`,
  poster: `img/snatch.jpg`,
  picture: `img/snatch.jpg`,
  id: 123891,
  description: `film description.`,
  rating: 8.3,
  scores: 1500,
  director: `Director Name`,
  starring: [`Author`],
  preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  filmDurationTime: 107,
  videoUrl: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  isFavorite: false,
  backgroundColor: `#FFFFFF`,
};

export const reviews: Array<ReviewType> = [{
  id: 1234567891,
  user: {
    id: 1223,
    name: `Ivan Ivanov`,
  },
  rating: 8.0,
  comment: `My comment.`,
  date: `2020-07-09T10:12:45.467Z`,
},
{
  id: 1234567891,
  user: {
    id: 1223,
    name: `Ivan Ivanov`,
  },
  rating: 8.0,
  comment: `My comment`,
  date: `2020-07-09T10:12:45.467Z`,
},
{
  id: 1234567891,
  user: {
    id: 1223,
    name: `Ivan Ivanov`,
  },
  rating: 8.0,
  comment: `My comment`,
  date: `2020-07-09T10:12:45.467Z`,
}];

export const genres: Array<string> = [
  `All genres`,
  `Drama`,
  `Thriller`,
  `Comedy`
];
