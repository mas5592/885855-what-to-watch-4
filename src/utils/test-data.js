export const card = {
  title: `Name`,
  genre: `film Genre`,
  date: 1993,
  coverBackground: `img/bg-the-grand-budapest-hotel.jpg`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  picture: `img/snatch.jpg`,
  id: 123456,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege`,
  rating: 8.9,
  ratingDescription: `Very good`,
  scores: 240,
  director: `Director Name`,
  starring: [`Author`],
  preview: `https://upload.wikimedia.org/wikipedia/commons/7/72/Landwasserviadukt%2C_aerial_video.webm`,
  filmDurationTime: 107,
  videoUrl: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  isFavorite: false,
  backgroundColor: `#FFFFFF`,
};

export const films = [
  {
    title: `Snatch`,
    genre: `Genre`,
    date: 2014,
    coverBackground: `https://placeimg.com/1300/512/nature`,
    poster: `img/snatch.jpg`,
    picture: `img/snatch.jpg`,
    id: 123465,
    description: `film description.`,
    rating: 8.3,
    scores: 1500,
    director: `Director Name`,
    starring: [`Author`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    filmDurationTime: 107,
    videoUrl: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    isFavorite: false,
    backgroundColor: `#FFFFFF`,
  }, {
    title: `Snatch`,
    genre: `Genre`,
    date: 2014,
    coverBackground: `https://placeimg.com/1300/512/nature`,
    poster: `img/snatch.jpg`,
    picture: `img/snatch.jpg`,
    id: 123478,
    description: `film description.`,
    rating: 8.3,
    scores: 1500,
    director: `Director Name`,
    starring: [`Author`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    filmDurationTime: 107,
    videoUrl: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    isFavorite: false,
    backgroundColor: `#FFFFFF`,
  }
];

export const activeFilm = {
  title: `Snatch`,
  genre: `Genre`,
  date: 2014,
  coverBackground: `https://placeimg.com/1300/512/nature`,
  poster: `img/snatch.jpg`,
  picture: `img/snatch.jpg`,
  id: 123891,
  description: `film description.`,
  rating: 8.3,
  scores: 1500,
  director: `Director Name`,
  starring: [`Author`],
  preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  filmDurationTime: 107,
  videoUrl: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  isFavorite: false,
  backgroundColor: `#FFFFFF`,
};

export const filmReviews = [{
  id: 1234567891,
  user: {
    id: 1223,
    title: `Ivan Ivanov`,
  },
  rating: 8.0,
  comment: `My comment`,
  date: `2020-07-09T10:12:45.467Z`,
},
{
  id: 1234567891,
  user: {
    id: 1223,
    title: `Ivan Ivanov`,
  },
  rating: 8.0,
  comment: `My comment`,
  date: `2020-07-09T10:12:45.467Z`,
},
{
  id: 1234567891,
  user: {
    id: 1223,
    title: `Ivan Ivanov`,
  },
  rating: 8.0,
  comment: `My comment`,
  date: `2020-07-09T10:12:45.467Z`,
}];

export const genres = [`All genres`, `Drama`, `Thriller`, `Comedy`];
