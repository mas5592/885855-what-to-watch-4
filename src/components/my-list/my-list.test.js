import React from 'react';
import {Router} from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import NameSpace from '../../reducer/name-space';
import MyList from './my-list';
import {films} from '../../utils/test-data.js';
import {AuthorizationStatus, AVATAR_URL} from '../../const';
import history from '../../history';

const mockStore = configureStore([]);

describe(`MyList`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      favoriteFilms: films,
    },
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.AUTH,
      userInfo: {
        id: 1,
        email: `ivan@dmail.ru`,
        name: `Ivan`,
        avatarUrl: AVATAR_URL,
      }
    }
  });

  it(`Should MyList render correctly`, () => {
    const tree = renderer
      .create(
          <Router history={history}>
            <Provider store={store}>
              <MyList />
            </Provider>
          </Router>, {
            createNodeMock: () => {
              return {};
            }
          })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
