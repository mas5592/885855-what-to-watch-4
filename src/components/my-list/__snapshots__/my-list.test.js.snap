// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyList Should MyList render correctly 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <a
        href="/mylist"
        onClick={[Function]}
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="Ivan"
            height={63}
            src="https://4.react.pages.academy/wtw/asda.jpg"
            width={63}
          />
        </div>
      </a>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          className="small-movie-card__link"
          href="/film/123465"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              className="player__video"
            />
            <img
              height={175}
              src="img/snatch.jpg"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            Snatch
          </h3>
        </a>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          className="small-movie-card__link"
          href="/film/123478"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              className="player__video"
            />
            <img
              height={175}
              src="img/snatch.jpg"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            Snatch
          </h3>
        </a>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
